### ISSUE TITLE
`next dev` on Windows crashes 

### ISSUE THREAD

**Issue by littledivy** [denoland/deno#22189]
To reproduce:
- `npx create-next-app` && cd app
-  Update `module.exports` to `export default` in next.config.js
- Add `deno.json`: `{ unstable: ["fs", "byonm"] }`
- `deno task dev`

```
PS C:\Users\dhairy\Desktop\divy\gh\my-app> deno task dev
Task dev next dev
   ▲ Next.js 14.0.4
   - Local:        http://localhost:3000

'npm' is not recognized as an internal or external command,
operable program or batch file.
 ✓ Ready in 1898ms


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 00000074B4388ED0
==== C stack trace ===============================

        napi_release_threadsafe_function [0x00007FF7222816CB+27318058]
        napi_release_threadsafe_function [0x00007FF722280717+27314038]
        napi_release_threadsafe_function [0x00007FF7222DD81D+27695228]
        napi_release_threadsafe_function [0x00007FF7222EA4F1+27747664]
        napi_release_threadsafe_function [0x00007FF7222EC1C6+27755045]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF7226CAEBD+2753181]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF7226CA536+2750742]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF7226C954F+2746671]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF7226C98CB+2747563]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF7226C968F+2746991]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF722B941F4+7771604]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF72327337A+14976858]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF7232FCD49+15540521]
        (No symbol) [0x00007FF6A38ACBAC]
error: Uncaught (in promise) TypeError: Unknown signal: 0
    at toDenoSignal (ext:deno_node/internal/child_process.ts:282:11)
    at ChildProcess.kill (ext:deno_node/internal/child_process.ts:196:53)
    at handleSessionStop (file:///C:/Users/dhairy/Desktop/divy/gh/my-app/node_modules/next/dist/cli/next-dev.js:82:15)
    at ChildProcess.<anonymous> (file:///C:/Users/dhairy/Desktop/divy/gh/my-app/node_modules/next/dist/cli/next-dev.js:267:23)
    at ChildProcess.emit (ext:deno_node/_events.mjs:383:28)
    at ext:deno_node/internal/child_process.ts:180:16
    at eventLoopTick (ext:core/01_core.js:63:7)
```

There 3 errors here:

1. `'npm' is not a recognised command` even though It is in my PATH
2. A C++ stacktrace of a Node-API module. Note that Deno would not produce a stack trace like this, this is likely a Node.js process.
3.` Unknown signal: 0` from our child_process kill implementation

**Comment by littledivy** [comment]
I just got the most absurd error message:

```
operable program or batch file.
 ✓ Ready in 2.1s
 ✓ Compiled /favicon.ico in 468ms (44 modules)
<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Unable to snapshot resolve dependencies


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0000003EB7F89170
==== C stack trace ===============================

        onig_get_syntax [0x00007FF729E835FB+7079647]
        onig_get_syntax [0x00007FF729E82647+7075627]
        onig_get_syntax [0x00007FF729EDF74D+7456817]
        onig_get_syntax [0x00007FF729EEC4A1+7509381]
        onig_get_syntax [0x00007FF729EEE176+7516762]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF72A2CCE6D+2753181]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF72A2CC4E6+2750742]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF72A2CB4FF+2746671]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF72A2CB87B+2747563]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF72A2CB63F+2746991]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF72A7961A4+7771604]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF72AE7533A+14976874]
        CrashForExceptionInNonABICompliantCodeRange [0x00007FF72AEFED09+15540537]
        (No symbol) [0x00007FF6AB458EEC]
error: Uncaught (in promise) TypeError: Unknown signal: 0
    at toDenoSignal (ext:deno_node/internal/child_process.ts:282:11)
    at ChildProcess.kill (ext:deno_node/internal/child_process.ts:196:53)
    at handleSessionStop (file:///C:/Users/dhairy/Desktop/divy/gh/demo/node_modules/next/dist/cli/next-dev.js:82:15)
    at ChildProcess.<anonymous> (file:///C:/Users/dhairy/Desktop/divy/gh/demo/node_modules/next/dist/cli/next-dev.js:267:23)
    at ChildProcess.emit (ext:deno_node/_events.mjs:383:28)
    at ext:deno_node/internal/child_process.ts:180:16
    at eventLoopTick (ext:core/01_core.js:64:7)
```

`onig_get_syntax`? What is a Ruby API function doing here?

**Comment by littledivy** [comment]
This has been fixed by #22247
